In [30]:
import numpy as np
num = []
n = 10
for i in range(30000):
    a = np.random.normal(0, 1, (n, n))
    A = a @ a.T
    mini = np.min(np.linalg.eigvalsh(A))
    A1 = A
    nu = 0
    while (abs(np.min(np.diag(A1))/mini - 1) > 0.01):
        nu += 1
        Q, R = np.linalg.qr(A1)
        A1 = R @ Q
    num.append(nu)
print(np.mean(num))

2.4561


Можно проверить (я так и сделал), что число итераций не зависит от n, однако, в силу использования функции random, количество итераций меняется, поэтому решено взять среднее для оценки. По итогу получается около 2.5 итераций, чтобы достаточно точно оценитьсобственные значения<br>